annotation-web/src/main/java/eu/europeana/annotation/web/service/controller/jsonld/WebAnnotationSearchRest.java

```diff
@@ -134,30 +147,47 @@ private ResponseEntity<String> searchAnnotation(String wskey, String queryString
 	}
     }
 
-    private SearchProfiles getProfile(String profile, HttpServletRequest request) throws ParamValidationI18NException, ParamValidationI18NException {
-
-	// if the profile parameter is given, the header preference is ignored
-	if (profile != null) {
-	    if (!SearchProfiles.contains(profile)) {
-	        throw new ParamValidationI18NException(I18nConstants.INVALID_PARAM_VALUE,
-				I18nConstants.INVALID_PARAM_VALUE,
-				new String[] { WebAnnotationFields.PARAM_PROFILE, profile });
-	    }
-	    
-	    return SearchProfiles.getByStr(profile);
-	}
-
-	String preferHeader = request.getHeader(HttpHeaders.PREFER);
-	if (preferHeader != null) {
-	    preferHeader = preferHeader.replaceAll("\\s+", "");
-	    if (preferHeader.equals(AnnotationHttpHeaders.VALUE_PREFER_CONTAINEDIRIS)) {
-		return SearchProfiles.MINIMAL;
-	    } else if (preferHeader.equals(AnnotationHttpHeaders.VALUE_PREFER_CONTAINEDDESCRIPTIONS)) {
-		return SearchProfiles.STANDARD;
-	    }
-	}
-
-	logger.trace("STANDARD Profile set by default");
-	return SearchProfiles.STANDARD;
+    private void validateProfiles(List<String> profiles) throws HttpException {
+      // For now maximum two profile-combinations are possible
+      // profile=facets OR profile=facets,minimal OR profile=standard,facets
+      if(profiles.size() > 2) {
+          throw new ParamValidationException(I18nConstants.INVALID_PARAM_VALUE, I18nConstants.INVALID_PARAM_VALUE,
+                  new String[]{"Maximum 2 profiles from these are allowed ", StringUtils.join(profiles, ",")});
+      }
+      // For now - if multiple profile then one of them has to be facets
+      if (profiles.size() == 2 && !profiles.contains(SearchProfiles.FACETS.toString())) {
+          throw new ParamValidationException(I18nConstants.INVALID_PARAM_VALUE, I18nConstants.INVALID_PARAM_VALUE,
+                  new String[]{"These profiles are not supported together ", StringUtils.join(profiles, ",")});
+      }
+      //check the profile names
+      for(String profile : profiles) {
+        if(!SearchProfiles.contains(profile)) {
+          throw new ParamValidationI18NException(I18nConstants.INVALID_PARAM_VALUE,
+              I18nConstants.INVALID_PARAM_VALUE,
+              new String[] { WebAnnotationFields.PARAM_PROFILE, profile });
+        }
+      }
+  }
+
+    private SearchProfiles getProfile(String profile, HttpServletRequest request) throws HttpException {
+      List<String> allProfiles = GeneralUtils.splitStringIntoList(profile, WebAnnotationModelFields.COMMA);
+      validateProfiles(allProfiles);      
+      allProfiles.remove(SearchProfiles.FACETS.toString());
+      if(allProfiles.size()>0) {
```

Comment:
we should take care of profile=debug here ... the method should return a list of profiles ...